File: src/main/java/org/goafabric/jobrunr/configuration/JobRunrConfiguration.java
package org.goafabric.jobrunr.configuration;

import org.jobrunr.spring.autoconfigure.JobRunrProperties;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JobRunrConfiguration {
    public JobRunrConfiguration(JobRunrProperties jobRunrProperties, @Value("${spring.autoconfigure.exclude:}") String autoConfigureExclude) {
        jobRunrProperties.getDatabase().setSkipCreate(!autoConfigureExclude.contains("DataSourceAutoConfiguration"));
    }
}

File: src/main/kotlin/org/goafabric/jobrunr/configuration/JobRunrConfiguration.kt
package org.goafabric.jobrunr.configuration

import org.jobrunr.spring.autoconfigure.JobRunrProperties
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
class JobRunrConfiguration(
    jobRunrProperties: JobRunrProperties,
    @Value("\${spring.autoconfigure.exclude:}") autoConfigureExclude: String
) {
    init {
        jobRunrProperties.database.isSkipCreate = !autoConfigureExclude.contains("DataSourceAutoConfiguration")
    }
}
